Observer/SaveOrderMarketingConsent.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Exception;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Helper\Webhook;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+
+class SaveOrderMarketingConsent implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * @var Webhook $webhookHelper
+     */
+    protected  $_webhookHelper;
+
+    /**
+     * @param Webhook $webhookHelper
+     * @param ScopeSetting $klaviyoScopeSetting
+     */
+    public function __construct(
+        Webhook $webhookHelper,
+        ScopeSetting $klaviyoScopeSetting
+    ) {
+        $this->_webhookHelper = $webhookHelper;
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+    }
+
+    /**
+     * customer register event handler
+     *
+     * @param Observer $observer
+     * @return void
+     * @throws Exception
+     */
+    public function execute(Observer $observer)
+    {
+        $order = $observer->getEvent()->getOrder();
+        $quote = $observer->getEvent()->getQuote();
+
+        $order->setData("kl_sms_consent", json_encode($quote->getKlSmsConsent()));
+        $order->setData("kl_email_consent", json_encode($quote->getKlEmailConsent()));
+
+        $shippingInfo = $quote->getShippingAddress();
+        $webhookSecret = $this->_klaviyoScopeSetting->getWebhookSecret();
+        $updatedAt = $quote->getUpdatedAt();
+
+        $data = array("data" => array());
+
+        if (
+            $webhookSecret
+            && $this->_klaviyoScopeSetting->getConsentAtCheckoutSMSIsActive()
+        ) {
+            $data["data"][] = array(
```

Comment:
are the extra empty brackets so that we nest the data under `0` ?

Reply:
just how to append to an array